### PR TITLE
fix(rust): Declare the correct compatibility with older Rust versions

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

This fixes an unnecessary build failure on Rust 1.66 and 1.67.

### Specifications

https://doc.rust-lang.org/cargo/reference/manifest.html#the-rust-version-field

## Solution

- Declare the actual incompatible Rust version
- Document that no specific rust version is supported except stable, don't tempt users to depend on earlier versions

## Review

This should probably make it in the stable release, because it's a serious usability bug for users on older compiler versions.

### Reviewer Checklist

  - [ ] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [ ] Are the PR labels correct?
  - [ ] Does the code do what the ticket and PR says?
    - [ ] Does it change concurrent code, unsafe code, or consensus rules?
  - [ ] How do you know it works? Does it have tests?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,11 +67,10 @@ Building Zebra requires [Rust](https://www.rust-lang.org/tools/install),
 [pkg-config](http://pkgconf.org/), and a C++ compiler.
 
 Zebra is tested with the latest `stable` Rust version. Earlier versions are not
-supported or tested. Note that Zebra's code currently uses features introduced
-in Rust 1.68, or any later stable release.
+supported or tested. Any Zebra version can start depending on new features in the
```
Comment (arya2) [inline README.md]:
```suggestion
supported or tested. Any Zebra release can start depending on new features in the
```

**Review by arya2 — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (teor2345) [reply README.md]: I used a slightly shorter wording.

**Comment by teor2345** [comment]
I tested this using:
```sh
rustup toolchain install 1.68 1.67 1.66 1.65
cargo +1.65 check
# failed as expected
cargo +1.66 check
# succeeded
cargo +1.66 build --ignore-rust-version --bin zebrad
#succeeded
```

**Review by arya2 — APPROVED** [review]

**Review by daira — APPROVED** [review]
ACK
